src/test/java/org/jenkinsci/plugins/workflow/steps/CatchErrorStepTest.java

```diff
@@ -234,6 +234,22 @@
         assertCatchError(r, b, Result.FAILURE, Result.FAILURE, true);
     }
 
+    @Issue("JENKINS-60239")
+    @Test public void parallelStageFails() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(""
+                + "node {\n"
+                + "    parallel(\n"
+                + "        stage1: { error 'fail' },\n"
+                + "        stage2: { catchError(stageResult: 'FAILURE', catchInterruptions: true) { sleep 10 } },\n"
```

Comment:
I think you can make this change in your Pipeline to get the desired behavior:

```suggestion
                + "        stage2: { catchError(stageResult: 'FAILURE', catchInterruptions: false) { sleep 10 } },\n"
```